scripts/common/preflights.sh

```diff
@@ -605,3 +605,26 @@ function host_preflights_mkresults() {
     chmod -R +r "${DIR}/${HOST_PREFLIGHTS_RESULTS_OUTPUT_DIR}/" # make sure the file is readable by kots support bundle
     rm -f "${tmp_file}"
 }
+
+# Uninstall Docker when containerd is selected to be installed and it is a new install
+# So that, is possible to avoid conflicts
+allow_remove_docker_new_install() {
+     # If docker is not installed OR if conatinerd is not in the spec
+     # then, the docker should not be uninstalled
+     if ! commandExists docker || [ -z "$CONTAINERD_VERSION" ]; then
+          return
+     fi
+
+     # if k8s is installed already then, the docker should not be uninstalled
+     # so that it can be properly migrated to containerd
+     if commandExists kubectl ; then
+          return
+     fi
```

Comment:
I don't think checking the existence of `kubectl` command is enough to determine if this is a first install. We have this logic in `reporting.sh` to determine if there was a previous kURL installation: https://github.com/replicatedhq/kURL/blob/main/scripts/common/reporting.sh#L29

Reply:
Working on that :-)

Reply:
Done and tested worked fine.